gateway-js/src/index.ts

```diff
@@ -469,27 +478,13 @@ export class ApolloGateway implements GraphQLService {
         "A valid schema couldn't be composed. Falling back to previous schema.",
       );
     } else {
-      this.schema = schema;
-      this.queryPlanner = new QueryPlanner(schema);
-
-      // Notify the schema listeners of the updated schema
-      try {
-        this.onSchemaChangeListeners.forEach((listener) =>
-          listener(this.schema!),
-        );
-      } catch (e) {
-        this.logger.error(
-          "An error was thrown from an 'onSchemaChange' listener. " +
-            'The schema will still update: ' +
-            ((e && e.message) || e),
-        );
-      }
+      this.updateWithSchemaAndNotify(schema, supergraphSdl);
```

Comment:
Yep, I'm down to move it into `updateWithSchemaAndNotify()`. We weren't doing it before in `loadStatic()`, but that's probably because it's only called at initial schema load, so it just ends up being a no-op.

I agree that "schema derived data" would be nice in gateway, although I'll leave that to future PRs.